pybamm/parameters/parameter_values.py

```diff
@@ -323,6 +324,29 @@ def update(self, values, check_conflict=False, check_already_exists=True, path="
                     # Save name and data
                     self._dict_items[name] = (function_name, data)
                     values[name] = (function_name, data)
+
+                # Machine learning data from Random Forest Regressor
+                elif value.startswith("[current 2D data]") or value.startswith("[2D data]"):
+                    if value.startswith("[current 2D data]"):
+                        data_path = os.path.join(
+                            pybamm.root_dir(), "pybamm", "input", "drive_cycles"
+                        )
+                        filename = os.path.join(data_path, value[17:] + ".json")
+                        function_name = value[17:]
```

Comment:
What is the use case for current 2D data? Won't it always be a function of time only?

Reply:
To be honest I am not 100% sure what the `current data` does. I just copied the code above for 1D and adapted it for my purposes.

Reply:
It looks to me like the `current data` uses a different root path when searching for the file from what I see from the source code...

Reply:
Yeah, since current profiles don't need to be associated with a particular chemistry unlike other parameters. You can get rid of this branch